libs/shared/profile-form/src/lib/shared-profile-form.tsx

```diff
@@ -0,0 +1,258 @@
+          />
+        </Box>
+        <Box my={1}>
+          <InputLabel id="photoLabel">Photo</InputLabel>
+          <TextField
+            id="photo"
+            name="photo"
+            type="file"
+            variant="outlined"
+            inputRef={register()}
+            fullWidth
+          />
+        </Box>
+        <Box my={1}>
+          <InputLabel id="fieldOfProfessionLabel">
+            Field of Profession
+          </InputLabel>
+          <TextField
+            id="fieldOfProfession"
+            name="fieldOfProfession"
+            type="text"
+            variant="outlined"
+            inputRef={register()}
+            fullWidth
+          />
+        </Box>
+        <Box my={1}>
+          <InputLabel id="companyLabel">Company</InputLabel>
+          <TextField
+            id="company"
+            name="company"
+            type="text"
+            variant="outlined"
+            inputRef={register()}
+            fullWidth
+          />
+        </Box>
+        <Box my={1}>
+          <FormControl variant="outlined" fullWidth>
+            <InputLabel id="categoriesLabel">Categories *</InputLabel>
+            <Controller
+              name="categories"
+              defaultValue={[]}
+              control={control}
+              onChange={([selected]) => {
+                return selected;
+              }}
+              as={
+                <Select
+                  id="categories"
+                  name="categories"
+                  multiple
+                  label="Categories"
+                  value={categories}
+                  onChange={handleChange}
+                  input={<Input id="select-multiple-chip" />}
+                  inputRef={register()}
+                  renderValue={(selected) => (
+                    <div>
+                      {(selected as string[]).map((value) => (
+                        <Chip key={value} label={value} />
+                      ))}
+                    </div>
+                  )}
+                >
+                  {props.categories.map((category) => (
+                    <MenuItem key={category.id} value={category.name}>
+                      {category.name}
+                    </MenuItem>
+                  ))}
+                </Select>
+              }
+              fullWidth
+            />
+          </FormControl>
+        </Box>
+        <Box my={1}>
+          <FormControl variant="outlined" fullWidth>
+            <InputLabel id="citiesLabel">City</InputLabel>
+            <Controller
+              name="city"
+              defaultValue=""
+              control={control}
+              onChange={([selected]) => {
+                return selected;
+              }}
+              as={
+                <Select
+                  labelId="city-label"
+                  id="city"
+                  name="city"
+                  value={city}
+                  onChange={(event) => setCity(event.target.value as string)}
```

Comment:
don't use 'as'

Reply:
Removing it gives me an error
`Argument of type 'unknown' is not assignable to parameter of type 'SetStateAction<string>'.
  Type 'unknown' is not assignable to type '(prevState: string) => string'.
    Type '{}' provides no match for the signature '(prevState: string): string'.ts(2345)`